EpsilonAuxBMS/Tests/tests/UpdateChargeAllowanceTaskTest/UpdateChargeAllowanceTaskTest.c

```diff
@@ -0,0 +1,178 @@
+#include "UpdateChargeAllowanceTaskTest.h"
+#include "unity.h"
+
+#include "Mockcmsis_os.h"
+#include "Mockstm32f4xx_hal_gpio.h"
+#include "UpdateChargeAllowanceTask.h"
+#include "AuxStatus.h"
+#include "MockTrip.h"
+#include "MockSetContactorsTask.h"
+#include "OrionStatusHelper.h"
+#include "stdio.h"
+
+void runUpdateChargeAllowanceTaskTests()
+{
+    RUN_TEST(test_UpdateChargeAllowanceTask_StandardOperationShouldUpdateAuxAndOrionStatusCorrectly);
+    RUN_TEST(test_UpdateChargeAllowanceTask_NoDischargeSenseShouldTurnOffPinAndUpdateOrionStatus);
+    RUN_TEST(test_UpdateChargeAllowanceTask_NoChargeSenseShouldTurnOffPinAndUpdateAuxAndOrionStatus);
+    RUN_TEST(test_UpdateChargeAllowanceTask_ShouldTripWillDisconnectContactorsAndUpdateAuxAndOrionStatus);
+    RUN_TEST(test_UpdateChargeAllowanceTask_TooHighMaxCellVoltageShouldTurnOffChargeAndUpdateStatuses);
+    RUN_TEST(test_UpdateChargeAllowanceTask_TooLowMinCellVoltageShouldTurnOffDischargeAndUpdateStatuses);
+}
+
+void setUpdateChargeAllowanceOsExpects(uint32_t* prevWakeTime)
+{
+    osDelayUntil_ExpectAndReturn(prevWakeTime, 100, osOK);
```

Comment:
Instead of `100` should we change it so that the variable that indicates the frequency in the task source file is used here? Might have to pull the variable into the header file